src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -162,7 +162,7 @@ class FieldOperatorTypeDeduction(traits.VisitorWithSymbolTableTrait, NodeTransla
     """
 
     @classmethod
-    def apply(cls, node: foast.FunctionDefinition) -> foast.FunctionDefinition:
+    def apply(cls, node: foast.FunctionDefinition | foast.LocatedNode) -> foast.FunctionDefinition:
```

Comment:
if you look at `test_copy.py` in test_execution, this `apply` gets accessed with both foast.FunctionDefinition and foast.FieldOperator

Reply:
I would suggest to create a `FoastNodeT` type var bound on `foast.LocatedNode` and then typing this `... node: FoastNodeT) -> FoastNodeT:`. Because that is really what it does.

Reply:
Though as far as I understand, `apply `should return `foast.FunctionDefinition`

mypy also complains about it if changed to what you suggested

Reply:
That's a misunderstanding. `apply` should return exactly the same type of FOAST node as you input. It just calls the visitor functions and all of them adhere to that, as far as I can see. I thought `FoastNodeT -> FoastNodeT` would be the way to tell mypy that.

For me locally it accepts it too.